pyiron_base/project/project_browser.py

```diff
@@ -0,0 +1,337 @@
+
+    def _update_pathbox(self, box):
+        path_color = '#DDDDAA'
+        home_color = '#999999'
+
+        def on_click(b):
+            if self._busy_check():
+                return
+            self._update_project(b.path)
+            self._busy_check(False)
+
+        buttons = []
+        tmppath = os.path.abspath(self.path)
+        if tmppath[-1] == '/':
+            tmppath = tmppath[:-1]
+        len_root_path = len(os.path.split(self._project_root_path[:-1])[0])
+        tmppath_old = tmppath + '/'
+        while tmppath != tmppath_old:
+            tmppath_old = tmppath
+            [tmppath, curentdir] = os.path.split(tmppath)
+            button = widgets.Button(description=curentdir + '/',
+                                    tooltip=curentdir,
+                                    layout=widgets.Layout(width='auto'))
+            button.style.button_color = path_color
+            button.path = tmppath_old
+            button.on_click(on_click)
+            if self.fix_path or len(tmppath) < len_root_path - 1:
+                button.disabled = True
+                if self._hide_path:
+                    button.layout.display = 'none'
+            buttons.append(button)
+        button = widgets.Button(icon="fa-home",
+                                tooltip=self._initial_project_path,
+                                layout=widgets.Layout(width='auto'))
+        button.style.button_color = home_color
+        button.path = self._initial_project
+        if self.fix_path:
+            button.disabled = True
+        button.on_click(on_click)
+        buttons.append(button)
+        buttons.reverse()
+        box.children = tuple(buttons)
+
+    def _on_click_file(self, filename):
+        filepath = os.path.join(self.path, filename)
+        self.output.clear_output(True)
+        try:
+            data = self.project[filename]
+        except(KeyError, IOError):
+            data = None
+        with self.output:
+            if data is not None and str(type(data)).split('.')[0] == "<class 'PIL":
+                try:
+                    data_cp = data.copy()
+                    data_cp.thumbnail((800, 800))
+                except:
+                    data_cp = data
+                display(data_cp)
+            elif data is not None:
+                display(data)
+            else:
+                print([filename])
+        if filepath in self._clickedFiles:
+            self._data = None
+            self._clickedFiles.remove(filepath)
+        else:
+            if data is not None:
+                self._data = FileData(data=data, file=filename, metadata={"path": filepath})
+            # self._clickedFiles.append(filepath)
+            self._clickedFiles = [filepath]
+
+    def _update_filebox(self, filebox):
+        # color definitions
+        dir_color = '#9999FF'
+        node_color = '#9999EE'
+        file_chosen_color = '#FFBBBB'
+        file_color = '#DDDDDD'
+
+        if self._node_as_dirs:
```

Comment:
This is a little hard to read. Alternatives:

Basically same length but more transparent:
```python
dirs, files = self.dirs, self.files
if self._node_as_dirs:
    dirs += self.nodes
else:
    files += self.nodes
```

Shorter! But maybe harder to read?
```python
dirs = self.dirs + self.nodes if self._nodes_as_dirs else self.dirs
files = self.files if self._nodes_as_dirs else self.files + self.nodes
```

Reply:
The first one does not work correctly `if not self._node_as_dirs`, it is duplicating the file list on each click. The second one works and will be used.